src/dags/parkeervakken.py

```diff
@@ -0,0 +1,357 @@
+import datetime
+import pathlib
+import re
+import dateutil
+import shapefile
+from airflow import DAG
+from airflow.operators.bash_operator import BashOperator
+from airflow.operators.postgres_operator import PostgresOperator
+from airflow.operators.python_operator import PythonOperator
+from environs import Env
+from shapely.geometry import Polygon
+from sqlalchemy import create_engine
+from swift_operator import SwiftOperator
+from common import default_args
+
+dag_id = "parkeervakken"
+
+# CONFIG = Variable.get(DAG_ID, deserialize_json=True)
+
+TABLES = dict(
+    BASE=f"{dag_id}_{dag_id}",
+    BASE_TEMP=f"{dag_id}_{dag_id}_temp",
+    REGIMES=f"{dag_id}_{dag_id}_regimes",
+    REGIMES_TEMP=f"{dag_id}_{dag_id}_regimes_temp",
+)
+
+
+SQL_CREATE_TEMP_TABLES = """
+    DROP TABLE IF EXISTS {{ params.base_table }}_temp;
+    CREATE TABLE {{ params.base_table }}_temp (
+        LIKE {{ params.base_table }} INCLUDING ALL);
+    ALTER TABLE {{ params.base_table }}_temp
+        ALTER COLUMN geom TYPE geometry(Polygon,28992)
+        USING ST_Transform(geom, 28992);
+    DROP TABLE IF EXISTS {{ params.base_table }}_regimes_temp;
+    CREATE TABLE {{ params.base_table }}_regimes_temp (
+        LIKE {{ params.base_table }}_regimes INCLUDING ALL);
+    DROP SEQUENCE IF EXISTS {{ params.base_table }}_regimes_temp_id_seq
+       CASCADE;
+    CREATE SEQUENCE {{ params.base_table }}_regimes_temp_id_seq
+        OWNED BY {{ params.base_table }}_regimes_temp.id;
+    ALTER TABLE {{ params.base_table }}_regimes_temp
+        ALTER COLUMN id SET DEFAULT
+        NEXTVAL('{{ params.base_table }}_regimes_temp_id_seq');
+"""
+
+SQL_RENAME_TEMP_TABLES = """
+    ALTER TABLE IF EXISTS {{ params.base_table }}
+        RENAME TO {{ params.base_table }}_old;
+    ALTER TABLE {{ params.base_table }}_temp
+        RENAME TO {{ params.base_table }};
+    ALTER TABLE IF EXISTS {{ params.base_table }}_regimes
+        RENAME TO {{ params.base_table }}_regimes_old;
+    ALTER TABLE {{ params.base_table }}_regimes_temp
+        RENAME TO {{ params.base_table }}_regimes;
+"""
+
+TMP_DIR = f"/tmp/{dag_id}"
+
+
+def import_data(shp_file, ids):
+    """
+    Import Shape File into database.
+    """
+    env = Env()
+    db_engine = create_engine(env("PARKEERVAKKEN_DB"))
```

Comment:
Ik twijfel nog hier: misschien kan deze vebinding via admin interface aangezet worden? @jjmurre

Reply:
Hier zou je idd de PostgresHook voor kunnen gebruiken, iets als:

            hook = PostgresHook(postgres_conn_id=postgres_conn_id)
            hook.run(<sql-code>)

De postgres_conn_id kun je evt. zelfs weglaten, dan wordt de 'postgres_default' gebruikt, dat is onze 'dataservices' database.

Reply:
Maar, ik vermoed dat de PARKEERVAKKEN_DB een andere is. Die zou dan idd als een extra Postgres connectie in Airflow kunnen worden gehangen en obv id worden gebruikt in de PostgresHook

Reply:
Thnx!